### DISCUSSION TITLE
😱 TypeError: Cannot assign to read only property ‘Symbol(Symbol.toStringTag)’ of object ‘#’ Helia error on nextJS. await createHelia() initialize libp2p RandomWalk error

### DISCUSSION THREAD

**Discussion by jojedev** [ipfs/helia#589] [General]
Hi there!, 
I’m just trying to do a basic setup with Helia in NextJS and I’m getting an RandomWalk error on initialization. I haven’t found anything related online.
so:

import { createHelia } from 'helia';
     try {
      const helia = await createHelia()
    }  catch (error) {
      console.error('Error uploading file to IPFS:', error);
    }
error returns:
DashboardTabs.tsx:21 Error uploading file to IPFS: TypeError: Cannot assign to read only property ‘Symbol(Symbol.toStringTag)’ of object ‘#’

 at new RandomWalk (random-walk.js:119:20)
    at new Libp2pNode (libp2p.js:328:47)
    at createLibp2pNode (libp2p.js:396:12)
    at async createLibp2p (index.js:48:18)
    at async createLibp2p (libp2p.js:40:18)
    at async createHelia (index.js:57:18)
    at async uploadToIPFS (DashboardTabs.tsx:19:21)

Before using Helia I was using ipfs-http-client, I had a big error in production and found out it was deprecated so I decided to go with Helia. I just wanted to upload a image to IPFS and get the CID. That’s it, just a basic implementation, Any advice would be appreciated!. :grin:

**Comment by SgtPooki** [comment]
Do you have a reproduction codebase? what version of nodejs are you using?
